docs/current_docs/cookbook/snippets/execute-concurrently/typescript/index.ts

```diff
@@ -0,0 +1,62 @@
+import { dag, Container, Directory, object, func } from "@dagger.io/dagger"
+
+@object()
+class MyModule {
+  /**
+   * Return the result of running unit tests
+   */
+  @func()
+  async test(source: Directory): Promise<string> {
+    return this.buildEnv(source)
+      .withExec(["npm", "run", "test:unit", "run"])
+      .stdout()
+  }
+
+  /**
+   * Return the result of running the linter
+   */
+  @func()
+  async lint(source: Directory): Promise<string> {
+    return this.buildEnv(source)
+      .withExec(["npm", "run", "lint"])
+      .stdout()
+  }
+
+  /**
+   * Return the result of running the type-checker
+   */
+  @func()
+  async typecheck(source: Directory): Promise<string> {
+    return this.buildEnv(source)
+      .withExec(["npm", "run", "typecheck"])
+      .stdout()
+  }
+
+  /**
+   * Run linter, type-checker, unit tests concurrently
+   */
+  @func()
+  async runAllTests(source: Directory): Promise<string> {
```

Comment:
@TomChv @helderco can you please help improve this example? In particular, error handling from the sub-functions?